rotkehlchen/globaldb/updates.py

```diff
@@ -602,49 +600,115 @@ def perform_update(
+        updates = self._retrieve_update_files(
+            local_schema_version=local_schema_version,
+            infojson=infojson,
+            up_to_version=up_to_version,
+        )
+
         with TemporaryDirectory() as tmpdirname:
             tmpdir = Path(tmpdirname)
+            temp_db_name = 'temp.db'
             connection, _ = initialize_globaldb(
                 global_dir=tmpdir,
-                db_filename='temp.db',
+                db_filename=temp_db_name,
                 sql_vm_instructions_cb=GlobalDBHandler().conn.sql_vm_instructions_cb,
             )
+
+            # use a critical section to avoid modifications in the globaldb during the update
+            # process since we ignore any possible change in the user globaldb once we started
+            # the update.
             with GlobalDBHandler().conn.critical_section():
                 # make sure global DB is not accessed anywhere else
                 _replace_assets_from_db(connection, global_db_path)
-            self._perform_update(
-                connection=connection,
-                assets_conflicts=conflicts,
-                local_schema_version=local_schema_version,
-                infojson=infojson,
-                up_to_version=up_to_version,
-            )
-            if len(self.conflicts) != 0:
-                # close the temporary DB and return the conflicts
+                self._perform_update(
+                    connection=connection,
+                    assets_conflicts=conflicts,
+                    up_to_version=up_to_version,
+                    updates=updates,
+                )
+                if len(self.conflicts) != 0:
+                    # close the temporary DB and return the conflicts
+                    connection.close()
+                    return [
+                        {'identifier': x[0].identifier, 'local': x[0].serialize(), 'remote': x[1].serialize()}  # noqa: E501
+                        for x in self.conflicts
+                    ]
+
+                # otherwise we are sure the DB will work without conflicts so let's
+                # now move the data to the actual global DB
                 connection.close()
-                return [
-                    {'identifier': x[0].identifier, 'local': x[0].serialize(), 'remote': x[1].serialize()}  # noqa: E501
-                    for x in self.conflicts
-                ]
-
-            # otherwise we are sure the DB will work without conflicts so let's
-            # now move the data to the actual global DB
-            connection.close()
-            connection = GlobalDBHandler().conn
-            with connection.critical_section():  # assure global DB is not accessed anywhere else
-                _replace_assets_from_db(connection, tmpdir / 'temp.db')
+                connection = GlobalDBHandler().conn
+                log.debug('Replace globaldb')
+                _replace_assets_from_db(connection, tmpdir / temp_db_name)
```

Comment:
I would pass `GlobalDBHandler().conn` directly, without assigning it to the `connection` variable first